### PR TITLE
Initial container service setup

### LINKED ISSUE #1: Hello, Container Service
_As an author of funcX functions I want there to be a service to build custom containers so I can offer functions that run across compute environments_

## Acceptance Criteria
1. New repo funcx_container_service
2. Basic flask app 
3. Docker file that runs flask app inside a uWSGI container
4. Initial pytests 
5. Git Actions to test, lint with flake8, and publish docker image
6. Apache licence
7. README

### PR DESCRIPTION
**Pull request by mtn**
- [x] New repo funcx_container_service
- [x] Basic flask app
- [x] Docker file that runs flask app inside a uWSGI container
- [x] Initial pytests
- [x] Git Actions to test, lint with flake8, and publish docker image
  - [x] Run tests
  - [x] Lint
  - [x] Publish image
- [x] Apache licence
- [x] README

Closes #1

### CODE REVIEW

**Comment by mtn** [comment]
I've put off publishing the docker image (needs secrets, and I figured I'd look into it a bit later anyways rather than doing something on the side and switching it). Other than that, I think we have the scaffolding we want -- I'm moving on to the next issue branching off this but let me know if there's something I should revisit :)

Edit: one thing in particular -- I didn't really try to make the docker image minimally small or anything, just copied it from the web service

File: funcx_container_service/application.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from flask import Flask
+
+app = Flask(__name__)
+
+
+@app.route("/environment")
```
Comment (BenGalewsky) [inline funcx_container_service/application.py]: I prefer the organization provided by [flask-restful library](https://flask-restful.readthedocs.io/en/latest/quickstart.html) where each REST endpoint is in its own class

File: funcx_container_service/application.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+def env_endpoint():
+    return "TODO :)"
+
+
+if __name__ == "__main__":
```
Comment (BenGalewsky) [inline funcx_container_service/application.py:5]: Go ahead and sort out mounting a config file into this app. Use the `APP_CONFIG_FILE` environment var to tell the path to load this config from (example in web service)

File: tests/routes/test_container_service.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+
+class TestContainerService:
+    def test_create_environment(self):
+        assert True
```
Comment (BenGalewsky) [inline tests/routes/test_container_service.py]: Use the [flask test_client](https://flask.palletsprojects.com/en/1.1.x/testing/#the-testing-skeleton) to invoke the REST endpoint.

**Review by BenGalewsky — CHANGES_REQUESTED** [review]
This is mostly there.. a couple of details that I should have put in the original ticket are commented upon in this review.

File: funcx_container_service/application.py (hunk 1 above)
Comment (ryanchard) [reply funcx_container_service/application.py]: I also like rest URIs to be plurals. POST /environments pushes a new one, GET /environments gets all, GET /environments/:id gets one in particular.

File: LICENSE [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [2019] [The University of Chicago]
```
Comment (ryanchard) [inline LICENSE]: I assume we need to update all of these to 2020?

File: funcx_container_service/application.py (hunk 1 above)
Comment (mtn) [reply funcx_container_service/application.py]: done

File: funcx_container_service/application.py (hunk 2 above)
Comment (mtn) [reply funcx_container_service/application.py:5]: I think this is done, but it's possible I misunderstood a piece -- just let me know :)

File: tests/routes/test_container_service.py (hunk 3 above)
Comment (mtn) [reply tests/routes/test_container_service.py]: done

**Review by BenGalewsky — APPROVED** [review]
Wonderful!

**Pull request merged**